src/react/AppProvider.jsx

```diff
@@ -65,11 +65,13 @@ export default function AppProvider({ store, children }) {
     <IntlProvider locale={locale} messages={getMessages()}>
       <ErrorBoundary>
         <AppContext.Provider
-          value={{ authenticatedUser, config, locale }}
+          value={
+            useMemo(() => ({ authenticatedUser, config, locale }), [authenticatedUser, config, locale])
```

Comment:
As a point of style, I'd prefer to remove this `useMemo` hook from the JSX and create an intermediate variable that we then pass into the `value` attribute here.  

```
const appContextValue = useMemo(() => ({ authenticatedUser, config, locale }), [authenticatedUser, config, locale]);
```

... That sorta thing.